src/persist-client/src/fetch.rs

```diff
@@ -386,28 +393,13 @@ where
     /// the returned `SerdeLeasedBatchPart` back into a `LeasedBatchPart` will
     /// leak `SeqNo`s and prevent persist GC.
     pub fn into_exchangeable_part(mut self) -> SerdeLeasedBatchPart {
-        let r = SerdeLeasedBatchPart {
-            shard_id: self.shard_id,
-            metadata: self.metadata.clone(),
-            lower: self.desc.lower().iter().map(T::encode).collect(),
-            upper: self.desc.upper().iter().map(T::encode).collect(),
-            since: self.desc.since().iter().map(T::encode).collect(),
-            key: self.part.key.clone(),
-            encoded_size_bytes: self.part.encoded_size_bytes,
-            leased_seqno: self.leased_seqno,
-            reader_id: self.reader_id.clone(),
-            stats: self.part.stats.clone(),
-            filter_pushdown_audit: self.filter_pushdown_audit,
-            key_lower: std::mem::take(&mut self.part.key_lower),
-            ts_rewrite: self
-                .part
-                .ts_rewrite
-                .as_ref()
-                .map(|x| x.iter().map(T::encode).collect()),
-        };
+        let (proto, _metrics) = self.into_proto();
```

Comment:
The way we're smuggling the metrics through here is a bit odd, but not a blocker IMO.

(Reading, it looks like the main reason we need it is for the `Drop` implementation, so it would be hard to remove from the leased part without a larger refactoring...)

Reply:
Yup, that's exactly it. I'll push up a comment and get a nightlies going on this!